### ISSUE TITLE
Cannot keyboard navigate right to Data grid Action Items when coming from the Header Cell

### ISSUE THREAD

**Issue by TheOneTheOnlyJJ** [mui/mui-x#17300]
### Search keywords

data grid keyboard key board nav navigation action cell header item right left arrow not cannot can icon come coming from

### Latest version

- [x] I have tested the latest version

### Affected products

Data Grid

### Steps to reproduce

Steps:
1. Open this link to live example: https://codesandbox.io/p/sandbox/actions-header-to-cell-navigation-bug-6kcfw5
2. Click any cell in the chart
3. Use keyboard navigation to navigate to the Actions column header cell
4. Use keyboard navigation again to navigate to the first Actions column row
5. Try using keyboard navigation again to navigate through the Action Items in the first Actions column row
6. Notice how it is not possible to do so. Pressing the "Right Arrow" does nothing. It is possible to navigate to other cells using the rest of the keys.

### Current behavior

The navigation action mentioned in point 5 has no effect.

### Expected behavior

With keyboard navigation, it should be possible to navigate between Action Cell Items when coming from the Actions Header Cell.

### Context

I randomly stumbled upon this problem while playing around with a Data Grid I made in my Bachelor's Thesis project.

Speaking of which...

@oliviertassinari 

I know this is absolutely not the place or proper way to do this, but I don't have any other options, so please hear me out.
I have tried reaching out at `sales@mui.com` but have not received a response. On X, I need to have a verified account to DM you. I've also looked online for other email addresses but have not found anything. So, here is my last-ditch effort to get a response.

I'm a final year student at the CS faculty of UPT (The Polytechnic University of Timisoara), and I'm currently working on my Bachelor's Thesis project (an electron.js app).
I use both Material UI and MUI X on the front end.
All the issues I've opened over the last year both here at MUI X and over at Material UI (quite a few, I would say) are related exclusively to my work on this project.

As my work progressed, however, I found the need to dip into the Pro and even Premium-only features of MUI X to deliver a high-quality project. I am heavily constrained by the 1 month only free trial.
The project must be finalised by mid-July. The 1-month free trial is not long enough to cover the final presentation as well. I have asked for the possibility of getting a 3-month free trial one way or another at `sales@mui.com` but got no response.

I decided to write this here because even if such a request cannot be met, I would at least like an answer so that I know how to proceed with my project, given the fact.

I hope this wall of text is not too big of a nuisance here, but I really don't know how to get an answer to this any other way.

### Your environment

<details>
  <code>System:
    OS: Windows 10 10.0.19045
  Binaries:
    Node: 20.18.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.8.2 - C:\Program Files\nodejs\npm.CMD   
    pnpm: Not Found
  Browsers:
    Chrome: 135.0.7049.41
    Edge: Chromium (127.0.2651.74)
  npmPackages:
    @emotion/react: ^11.14.0 => 11.14.0
    @emotion/styled: ^11.14.0 => 11.14.0
    @mui/core-downloads-tracker:  6.4.7
    @mui/icons-material: ^6.4.7 => 6.4.7
    @mui/material: ^6.4.7 => 6.4.7
    @mui/private-theming:  6.4.6
    @mui/styled-engine:  6.4.6
    @mui/system:  6.4.7
    @mui/types:  7.2.21
    @mui/utils:  6.4.6
    @mui/x-data-grid: ^7.27.3 => 7.27.3
    @mui/x-internals:  7.26.0
    @types/react: ^18.3.3 => 18.3.3
    react: ^18.3.1 => 18.3.1
    react-dom: ^18.3.1 => 18.3.1
    typescript: ^5.8.2 => 5.8.2</code>
</details>
